e2e/test/Crowdsale.test.ts

```diff
@@ -0,0 +1,708 @@
+import { ApiPromise, Keyring, WsProvider } from "@polkadot/api";
+import type { KeyringPair } from "@polkadot/keyring/types";
+import { DispatchError } from "@polkadot/types/interfaces";
+import { hexToU8a } from "@polkadot/util";
+import { expect } from "chai";
+import { Wallet } from "ethers";
+
+import { ALITH_PRIVATE_KEY, GAS_TOKEN_ID, NATIVE_TOKEN_ID, NodeProcess, finalizeTx, futurepassAddress, getNextAssetId, nftCollectionIdToCollectionUUID, startNode, typedefs } from "../common";
+
+describe("Crowdsale pallet", () => {
+  let node: NodeProcess;
+
+  let api: ApiPromise;
+  let alith: KeyringPair;
+
+  before(async () => {
+    node = await startNode();
+
+    // substrate variables
+    const wsProvider = new WsProvider(`ws://127.0.0.1:${node.wsPort}`);
+    api = await ApiPromise.create({ provider: wsProvider, types: typedefs });
+    alith = new Keyring({ type: "ethereum" }).addFromSeed(hexToU8a(ALITH_PRIVATE_KEY));
+  });
+
+  after(async () => await node.stop());
+
+  it("crowdsale extrinsic gas fees", async () => {
+    // TODO:
+  });
+
+  it("distribute crowdsale rewards cannot be called manually", async () => {
+    // validate signer cannot manually call distributeCrowdsaleRewards
+    const dispatchError = await new Promise<DispatchError>((resolve, reject) => {
+      api.tx.crowdsale
+        .distributeCrowdsaleRewards()
+        .signAndSend(alith, ({ status, dispatchError }) => {
+          if (!status.isFinalized) return;
+          if (dispatchError === undefined) return;
+          resolve(dispatchError);
+        })
+        .catch((err) => reject(err));
+    });
+    expect(dispatchError.isBadOrigin).to.be.true;
+  });
+
+  it("crowdsale - ROOT", async () => {
+    // crowdsale vars
+    const participants = Array.from({ length: 5 }, () => new Keyring({ type: "ethereum" }).addFromSeed(hexToU8a(Wallet.createRandom().privateKey))); // crowsale participants (10)
```

Comment:
nit, we aren't selling crows
```suggestion
    const participants = Array.from({ length: 5 }, () => new Keyring({ type: "ethereum" }).addFromSeed(hexToU8a(Wallet.createRandom().privateKey))); // crowdsale participants (10)
```

Reply:
https://github.com/futureversecom/trn-seed/pull/802/commits/c779f836f0d7016ca470eb3dca74d56a4273d33d